tensorflow_addons/layers/multihead_attention.py

```diff
@@ -189,6 +189,11 @@ def call(self, inputs, training=None, mask=None):
         key = inputs[1]
         value = inputs[2] if len(inputs) > 2 else key
 
+        if key.shape[-2] != value.shape[-2]:
+            raise ValueError(
+                "the number of elements in 'key' must be equal to the same as the number of elements in 'value'"
+            )
+
         # verify shapes
```

Comment:
we might also want to move this comment upwards

Reply:
done